notification/store.go

```diff
@@ -280,3 +295,45 @@ func (db *DB) VerifyContactMethod(ctx context.Context, cmID string, code int) er
 
 	return nil
 }
+
+func (db *DB) FindManyMessageStatuses(ctx context.Context, ids ...string) ([]MessageStatus, error) {
+	err := permission.LimitCheckAny(ctx, permission.User, permission.Admin)
+	if err != nil {
+		return nil, err
+	}
+	if len(ids) == 0 {
+		return nil, nil
+	}
+
+	err = validate.ManyUUID("IDs", ids, search.MaxResults)
+	if err != nil {
+		return nil, err
+	}
+
+	rows, err := db.findManyMessageStatuses.QueryContext(ctx, sqlutil.UUIDArray(ids))
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var result []MessageStatus
+	var s MessageStatus
+	for rows.Next() {
+		var lastStatus string
+		err = rows.Scan(&s.ID, &lastStatus, &s.Details, &s.ProviderMessageID, &s.Sequence)
+		if err != nil {
+			return nil, err
+		}
+
+		switch lastStatus {
+		case "delivered":
+			s.State = MessageStateDelivered
+		case "failed":
+			s.State = MessageStateFailedPerm
+		}
```

Comment:
```suggestion
		switch lastStatus {
		case "pending", "sending", "queued_remotely":
			s.State = MessageStateActive
		case "sent":
			s.State = MessageStateSent
		case "delivered":
			s.State = MessageStateDelivered
		case "failed", "bundled": // bundled message was not sent (replaced) and should never be re-sent
			// temporary if retry
			if hasNextRetry {
				s.State = MessageStateFailedTemp
			} else {
				s.State = MessageStateFailedPerm
			}
		default:
			return nil, fmt.Errorf("unknown last_status %s", lastStatus)
		}
```

Here are all the possible types for reference:
```
    'pending',
    'sending',
    'queued_remotely',
    'sent',
    'delivered',
    'failed',
    'bundled'
```

As for the `hasNextRetry` we can add a `next_retry_at notnull` to the SELECT query

Reply:
Should we pass this back to graphql transparently instead of mapping these to `ERROR` and `OK`? Then we can make more granular hints for our users (i.e. `Sent` vs `Delivered`, `Failed` vs `Failed. Retrying`, etc)